src/main/java/org/wildfly/security/auth/realm/IdentitySharedExclusiveLock.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2016 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.security.auth.realm;
+
+/**
+ * A simple shared/exclusive lock for a realm identity.
+ *
+ * @author <a href="mailto:fjuma@redhat.com">Farah Juma</a>
+ */
+public class IdentitySharedExclusiveLock {
+
+    private int sharedHoldCount;
+    private boolean isExclusiveLocked;
+    private int exclusiveRequests;
+    private Thread exclusiveThread = null;
+
+    /**
+     * Acquire the exclusive lock. An invocation of this method will block until the lock can be acquired
+     * or the invoking thread is interrupted.
```

Comment:
These docs don't match the semantics of the method, which continues to wait when the thread is interrupted.